### PR TITLE
recaptcha, preferences, :js, spellings

### LINKED ISSUE #7: Preferences data is not accessible
Upon new member sign up, they are asked to check off any of the following at the bottom of the registration page, but we do not get this information anyplace at our end.  Even in a CSV download, this information does not display.  Ideally, once someone asks for any of these things, we need to respond.  So we need some sort of notification system attached to it, or that shows on the Users list, or is included in te CSV download, which we do once a week to coordinate new members into  Constant Contact:
Send Me Signs Of Life Newsletter via email

Send Me Signs Of Life Newsletter via postal service (for residents of the US only)

I would like to be contacted about Volunteering my time to Forever Family Foundation

Please send me more information about Business Membership & Advertising Package

Please send me more information about becoming an Benefactor

Please contact me about making a Memorial Contribution

### LINKED ISSUE #67: New User Registration Confirmation
Confirm new user is not a bot either by confirmation email or authorisation image.
It is thought there are bots registering on the site

### LINKED ISSUE #74: Preferences: change newsletter to magazine, make delivery choice radio button
Delivery choice between "email" and "post" should be choose-one, "email" should be default

### PR DESCRIPTION
**Pull request by peterpih**
Add reCAPTCHA verification for registering new members
Set default testing to :js from :selenium
Show user preferences when viewing profile
Correct spelling Bussines -> Business
Closes #67, Closes #7, Closes #74

### CODE REVIEW

File: app/controllers/registration_controller.rb [hunk 1]
```diff
@@ -7,18 +7,23 @@ def new
...
-      flash[:notice] = I18n.t('flash.user.create.success')
-      sign_in(@user)
-      if @user.is_business?
-        redirect_to new_user_business_path(@user)
+    if verify_recaptcha
+      @user = User.new(params[:user])
+      if @user.save
+        flash[:notice] = I18n.t('flash.user.create.success')
+        sign_in(@user)
+        if @user.is_business?
```
Comment (dinshaw) [inline app/controllers/registration_controller.rb]: Three levels of conditionals is considered a code smell. Possible opportunity for some extraction of logic here.

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # This file is copied to spec/ when you run 'rails generate rspec:install'
 ENV["RAILS_ENV"] ||= 'test'
+
+# for reCAPTCHA testing
+ENV["RECAPTCHA_PRIVATE_KEY"] = "6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe"
```
Comment (dinshaw) [inline spec/rails_helper.rb]: not the place for config vars
see the 'figaro' gem.

Comment (peterpih) [inline spec/rails_helper.rb]: These keys are for testing(?), not the real keys. Real keys go into Heroku Settings, should these be taken out?

File: .gitignore [hunk 3]
```diff
@@ -27,3 +27,6 @@ public/cms_images
 public/system
 spec/support/env.rb
 tmp/
+
+# Ignore application configuration
```
Comment (dinshaw) [inline .gitignore]: not a show stopper, but this comment is unnecessary.

File: app/controllers/registration_controller.rb [hunk 4]
```diff
@@ -20,6 +20,10 @@ def create
       clean_up_passwords(@user)
       render '/users/new'
     end
+    # else
```
Comment (dinshaw) [inline app/controllers/registration_controller.rb]: what's up with the commented out code?

File: circle.yml [hunk 5]
```diff
@@ -1,3 +1,7 @@
+machine:
+  environment:
+    RECAPTCHA_PUBLIC_KEY: 6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI
```
Comment (dinshaw) [inline circle.yml]: sorry, still not the place for them. The goal is to keep all values of this sort out of the source control.
CircleCI has an interface to set env vars similar to heroku's.

File: config/initializers/recaptcha.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+Recaptcha.configure do |config|
+  config.public_key  = ENV['RECAPTCHA_PUBLIC_KEY']
+  config.private_key = ENV['RECAPTCHA_PRIVATE_KEY']
+  #config.api_version = 'v2'
```
Comment (dinshaw) [inline config/initializers/recaptcha.rb]: why the comment?

**Comment by dinshaw** [comment]
:+1:

**Pull request merged**